data_steward/validation/participants/identity_match.py

```diff
@@ -1,379 +1,336 @@
+    :param list_one: list of strings to check for existence in list_two
+    :param list_two: list of strings to use for existence check of
+        list_one parts
+    :return: the count of items in list_one that are missing from list_two
     """
-    return []
+    diff = 0
+    for part in list_one:
+        if part not in list_two:
+            diff += 1
+    return diff
 
 
-def _get_pii_emails(hpo):
+def _compare_name_fields(
+        project,
+        rdr_dataset,
+        pii_dataset,
+        hpo,
+        concept_id,
+        pii_field
+    ):
     """
-    Get email addresses from the site's PII table.
+    For an hpo, compare all first, middle, and last name fields to omop settings.
 
-    :param hpo:  hpo string to use when identifying table names for lookup.
+    This compares a site's name field values found in their uploaded PII
+    tables with the values in the OMOP observation table.
 
-    :return:  A list of tuples with the first tuple element as the person_id
-    and the second tuple element as the email address.
-    [(1, 'anon@gmail.com'), (99, '')]
+    :param hpo: string name of hop to search.
+    :return: a match_values dictionary.
     """
-    return []
+    match_values = {}
 
+    rdr_names = readers.get_rdr_match_values(
+        project, rdr_dataset, consts.ID_MATCH_TABLE, concept_id
+    )
 
-def _get_pii_phone_numbers(hpo):
-    """
-    Get email addresses from the site's PII table.
+    pii_names = readers.get_pii_values(
+        project,
+        pii_dataset,
+        hpo,
+        consts.PII_NAME_TABLE,
+        pii_field
+    )
 
-    :param hpo:  hpo string to use when identifying table names for lookup.
+    for person_id, pii_name in pii_names:
+        rdr_name = rdr_names.get(person_id)
 
-    :return:  A list of tuples with the first tuple element as the person_id
-    and the second tuple element as the phone number.
-    [(1, '5558675309'), (48, '5558004600'), (99, '5551002000')]
-    """
-    return []
+        if rdr_name is None or pii_name is None:
+            match_str = consts.MISSING
+        else:
+            pii_name = normalizer.normalize_name(pii_name)
+            rdr_name = normalizer.normalize_name(rdr_name)
+            match_str = consts.MATCH if rdr_name == pii_name else consts.MISMATCH
 
+        match_values[person_id] = match_str
 
-def _get_pii_addresses(hpo):
-    """
-    Get email addresses from the site's PII table.
+    return match_values
 
-    :param hpo:  hpo string to use when identifying table names for lookup.
 
-    :return:  A list of tuples with the first tuple element as the person_id,
-    the second tuple element as the address line one, the third tuple element is
-    address line 2, the fourth tuple element is city, the fifth tuple element is
-    state, and the sixth tuple element is zip code.
-    [(1, 'street 1', '', 'nashville', 'tn', 33333),
-     (2, 'street_8', 'street_10', 'yuma', 'az', 22222)]
+def _compare_email_addresses(
```

Comment:
doc strings for all the functions in this script needs to be updated with parameter descriptions.

Reply:
All doc strings in the module are updated.  Thanks for catching it.